addon/components/base/bs-button.js

```diff
@@ -449,4 +452,36 @@ export default class Button extends Component {
 
     return this.get('bubble');
   }
+
+  init() {
+    super.init(...arguments);
+
+    // deprecate arguments used for attribute bindings only
+    [
+      ['buttonType:type', 'submit'],
+      ['title', 'foo'],
+    ].forEach(([mapping, value]) => {
+      let argument = mapping.split(':')[0];
+      let attribute = mapping.includes(':') ? mapping.split(':')[1] : argument;
+      let deprecationMessage =
+        `Argument ${argument} of <BsButton> component is deprecated. It's only purpose ` +
+        `was setting the HTML attribute ${attribute} of the control element. You should use ` +
+        `angle bracket  component invocation syntax instead:\n` +
+        `Before:\n` +
+        `  {{bs-button ${attribute}="${value}"}}\n` +
+        `  <BsButton @${attribute}="${value}" />\n` +
+        `After:\n` +
+        `  <BsButton ${attribute}="${value}" />`;
+
+      deprecate(
```

Comment:
One "problem" here and at the other places where `deprecate()` is called, is that the preceding code add's a lot of unnecessary code to production builds. Ember (I guess ember-cli-babel) will remove any occurrence of `deprecate()` from production, but only that. So either
* put all code into a number of explicit `deprecate()` calls, i.e. duplication the message, the assertion etc., nothing else
* wrap everything related to deprecations in a `runInDebug` call, see https://api.emberjs.com/ember/3.14/functions/@ember%2Fdebug/runInDebug

Reply:
Good catch. Have wrapped the deprecation related code in `runInDebug` in f9df0e7d499710b54ee450fbd4c5a0d1c156d64e.